### PR TITLE
Clean version of Gaurab PR #639 : Unifying the way we report file related errors

### PR DESCRIPTION
**Pull request by epernod**
Unfortunately Gaurab seems to be unreachable those days. As I want to use his good work, here is a quick cherry pick of his PR #639

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-scene-tests]

File: SofaKernel/framework/sofa/helper/logging/Messaging.h [hunk 1]
```diff
@@ -130,7 +130,7 @@
 
 #define FILEINFO(filename, line) sofa::helper::logging::FileInfo(filename, line)
 
-#define FILEINFO(filename, line) sofa::helper::logging::FileInfo(filename, line)
+#define FILEINFO(filename, line) sofa::helper::logging::FileInfo(filename, line)  //rewritten again. delete this?
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/helper/logging/Messaging.h]: Maybe delete it :)

File: SofaKernel/framework/sofa/helper/io/MeshTrian.cpp [hunk 2]
```diff
@@ -54,13 +55,12 @@ void MeshTrian::init (std::string filename)
         return;
     }
     FILE *f = fopen(filename.c_str(), "r");
+    msg_error_when(!f)<<sofa::helper::messaging::FileMessage::UnableToOpenFile(filename.c_str());
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/helper/io/MeshTrian.cpp]: Unable -> unable

Comment (epernod) [reply SofaKernel/framework/sofa/helper/io/MeshTrian.cpp]: done in last commit ee41312

File: SofaKernel/framework/sofa/helper/logging/Messaging.h (hunk 1 above)
Comment (epernod) [reply SofaKernel/framework/sofa/helper/logging/Messaging.h]: done in last commit ee41312

**Review by guparan — APPROVED** [review]

**Comment by damienmarchal** [comment]
This one is a small PR so maybe we can merge it before the 7 days ? 
[ci-build][with-scene-tests]

**Comment by epernod** [comment]
I would say yes as it is the clean version of a 3 weeks old PR.

**Comment by gaurab4163** [comment]
I had been into competitive programming these days and also busy with a small project of mine. Thanks @epernod for making good use of my work. :smile:

**Comment by epernod** [comment]
you are welcome. Thanks for your work.

**Pull request merged**